### ISSUE TITLE
LSPS1: Let client tell LSP about onchain payment's

### ISSUE THREAD

**Issue by tnull** [BitcoinAndLightningLayerSpecs/lsp#45]
In LSPS1 the LSP returns a [`Payment`](https://github.com/BitcoinAndLightningLayerSpecs/lsp/blob/main/LSPS1/README.md#3-payment) object including a payable `onchain_address`.

Implementing this requires close integration between the on-chain and Lightning parts, in particular it will require that the LSP monitors the mempool and chain for any incoming payments. As this can be in some cases be rather complicated, it would be much easier if the client would tell us about the particular `txid` we should look out for. 

It's also noteworthy that mempool access cannot be assumed for all implementations so in some cases the `onchain_payments` array might be empty until the incoming transaction is confirmed.

**Comment by ZmnSCPxj-jr** [comment]
How is it complicated?  Electrum server interface lets you scan for specific transaction outputs with specific `scriptPubKey` (from an address you want to watch).  BIP158 includes both txid and `scriptPubKey` in the block filters.  Fullnode-based software can fetch recent blocks from `bitcoind` and scan individual transaction outputs.  What interface to the blockchain cannot scan `scriptPubKey`?

**Comment by SeverinAlexB** [comment]
I all for simplifying the spec but as Zmn mentioned, RBF is an issue. Is there any way to solve this issue?

**Comment by ZmnSCPxj-jr** [comment]
> as we'd need to track confirmation depth, handle reorgs etc.

Every on-chain wallet kinda *has to* do this regardless, because *everything is RBF* despite what 0-confers might like the world to be.

>In the LDK case the LSP client is really a module interfacing with LDK, which itself does not have integrated on-chain capabilities, neither for on-chain monitoring or the wallet parts. That means that the user needs to tell us about any change happening on-chain via corresponding interfaces, which currently don't allow to monitor incoming transactions to arbitrary `scriptpubkey`s, as it's not needed by Lightning.

Ummmmm the client is not the one who cares about the confirmation of the payment, it is the LSP that is the one who cares about that, right?  And the LSP has to open a channel, and opening  a channel requires that the LSP code --- outside of LDK ---- is able to construct and sign an onchain transaction going from its onchain funds to a channel outpoint.  So the LSP code needs *some* way to figure out when it receives onchain funds anyway (from e.g. the parent company of the LSP) so it seems to me not much more onerous to just use an address that would be monitored by the onchain wallet anyway?  Like if you have a mechanism to move funds from the parent company of the LSP to the LSP node itself so the LSP can actually fund channels, I expect the most standard way to do that is for the LSP to provide an address that the parent company will send money to, that same mechanism can be used to monitor the blockchain.

Even on the client side you need some mechanism to handle onchain funds since LSPS1 is primarily expected to be paid via onchain funds as well, how is the client going to get onchain funds to buy inbound capacity in the first place?

In short you need an onchain wallet anyway.  And an LSP might want to receive its funds in some kind of controlled address domain; for example, it might want to perform chain analysis if the onchain funds are from some kind of sanctioned address, so you need extra code outside of LDK anyway that checks the transaction is fine, and cancels the channel open if not, etc.  i.e. the interface to the LSPS1-LDK module should expose some kind of `give_lsps1_address` and `lsps1_address_was_funded` from the LSP code to the LSPS1-LDK module, and possibly `lsps1_address_failed` if the transaction that funded it failed some policy check that the LSP code imposes.
